components/badge/badge.stories.js

```diff
@@ -54,6 +54,15 @@ export const argTypesData = {
     },
     if: { arg: 'type', neq: 'ai' },
   },
+
+  decoration: {
+    defaultValue: 'default',
+    control: {
+      type: 'select',
+      options: Object.keys(BADGE_DECORATION_MODIFIERS),
+    },
+    // TODO: Is there any way to add conditions on more than one control?
```

Comment:
This prop is only valid when kind is label and type is default with no iconLeft and iconRight.
So it would have been better if we could have shown this only when the above conditions were met.
By writing something like
```
if: [
  { arg: 'kind', eq: 'label' },
  { arg: 'type', eq: 'default' },
  { arg: 'iconLeft', eq: '' },
  { arg: 'iconRight', eq: '' }
]
```
But that's not possible.